### PR TITLE
OnEntityClickListener implemented. Added max zoom (scale, in ESRI terms). Removed redundant IMapView interface. Removed redundant GGMap interface (combined it with GGMapView).

### PR DESCRIPTION
**Pull request by BarAgam**
OnEntityClickListener implemented.
Added max zoom (scale, in ESRI terms).
Removed redundant IMapView interface.
Removed redundant GGMap interface (combined it with GGMapView).

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 1]
```diff
@@ -89,7 +91,7 @@ public void updateMapEntity(GeoEntityNotification geoEntityNotification) {
 
     @Override
     public void setOnEntityClickedListener(MapEntityClickedListener mapEntityClickedListener) {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:93]: Not part of your new code, but what happens when we need more than one listener?
I think we should change it to AddOnEntityClickedListener.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 2]
```diff
@@ -180,12 +151,40 @@ private void onBasemapLoaded() {
         setOnStatusChangedListener(null);
     }
 
+    private void setExtentOverIsrael() {
+        Envelope israelEnvelope = new Envelope(
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:155]: You can make it simpler. This method can get Point as parameter.
Change the constants to one Point constants if possible.

**Review by teamagam3 — APPROVED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:93]: YAGNI

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java (hunk 2 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:155]: I cant :<
It needs to get the extent to set.
Am I missing something ?
I need 4 points for that  (envelope/4-sided polygon) how can I set the extent from a single point ?

**Pull request closed without merging**